### PR TITLE
fix: SiderMenu collapse for breakpoint change and make SiderMenu transition works.

### PR DESCRIPTION
**Pull request by Trendymen**

### CODE REVIEW

**Comment by Trendymen** [comment]
The Sider menu still has many problems.  includes the openKeys didn't be cache when collapsed and didn't restore when expanded.I'm also a little busy at my work, but i will try to help to resolve these.

File: src/SiderMenu/BaseMenu.tsx [hunk 1]
```diff
@@ -70,11 +71,15 @@ export const VueBaseMenuProps = {
 
 const httpReg = /(http|https|ftp):\/\/([\w.]+\/?)\S*/;
 
-const renderTitle = (title: string | undefined, i18nRender: FormatMessage) => {
-  return <span>{(i18nRender && title && i18nRender(title)) || title}</span>;
+const renderTitle = (title: string | undefined, i18nRender: FormatMessage, prefixCls: string) => {
+  return (
+    <span class={`${prefixCls}-item-title`}>
```
Comment (sendya) [inline src/SiderMenu/BaseMenu.tsx:76]: why add class in here

Comment (Trendymen) [reply src/SiderMenu/BaseMenu.tsx:76]: > why add class in here


just for following the implement of ant-design-pro. myable it look like redundant.
https://github.com/ant-design/pro-components/blob/51ea2173fb1fa60fe8ef30985807da5bade33ce8/packages/layout/src/components/SiderMenu/index.less#L86-L88

but what confused me is the class style already exist but never be used.
https://github.com/vueComponent/pro-layout/blob/1cb1789c649f1f7a8b1fcc6cc01264b0b13020a1/src/SiderMenu/index.less#L171-L174

Comment (Trendymen) [reply src/SiderMenu/BaseMenu.tsx:76]: so,I have a question, is't necessary reuse the less file completely from [pro-components](https://github.com/ant-design/pro-components/)? if it's, we can just copy those file then implement the render....

**Comment by sendya** [comment]
`BaseMenu` refactor render functional, this PR has expired

Thank your PR.

**Pull request closed without merging**